qemu/tests/rh_qemu_iotests.py

```diff
@@ -0,0 +1,155 @@
+import os
+import re
+import logging
+import shutil
+
+from virttest import utils_misc
+from virttest import error_context
+
+from avocado.utils import process
+from avocado.core import exceptions
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Get qemu src code from src rpm and run qemu-iotests using the
+    qemu binaries.
+
+    1) Get url of src rpm from brew
+    2) Unpack src code and apply patches
+    3) Run test for the file format detected
+    4) Check result
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+    def retry_command(cmd):
+        """
+        Retry the command when it fails, and raise the error once
+        retry has exceeded the maximum number
+
+        :param cmd: command expect to be executed
+        :return: output of the command
+        """
+        max_retry = int(params.get("max_retry", 3))
+        retry = max_retry
+        while retry:
+            retry -= 1
+            try:
+                return process.system_output(cmd, shell=True)
+            except process.CmdError, detail:
+                msg = "Fail to execute command"
+                logging.error("%s: %s." % (msg, detail))
+        raise exceptions.TestError("%s after %s times retry: %s" %
+                                   (msg, max_retry, detail))
+
+    def get_qemu_src_rpm_url():
```

Comment:
Not need this function, please use brew download it directly.
srpm=$(rpm -qf  /usr/libexec/qemu-kvm|sed 's/x86_64/src/').rpm; brew dowload-build --rpm "$srpm"

can help you download srpm directly.